server/src/main/java/org/elasticsearch/search/aggregations/pipeline/PercentileRanksBucketPipelineAggregationBuilder.java

```diff
@@ -0,0 +1,193 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.pipeline;
+
+import com.carrotsearch.hppc.DoubleArrayList;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.aggregations.AggregationBuilder;
+import org.elasticsearch.search.aggregations.AggregatorFactory;
+import org.elasticsearch.search.aggregations.PipelineAggregationBuilder;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+
+public class PercentileRanksBucketPipelineAggregationBuilder
+    extends BucketMetricsPipelineAggregationBuilder<PercentileRanksBucketPipelineAggregationBuilder> {
+    public static final String NAME = "percentile_ranks_bucket";
+    static final ParseField PERCENTS_FIELD = new ParseField("percents");
+    static final ParseField KEYED_FIELD = new ParseField("keyed");
+
+    private double[] percents = new double[] { 1.0, 5.0, 25.0, 50.0, 75.0, 95.0, 99.0 };
+    private boolean keyed = true;
+
+    public PercentileRanksBucketPipelineAggregationBuilder(String name, String bucketsPath) {
+        super(name, NAME, new String[] { bucketsPath });
+    }
+
+    /**
+     * Read from a stream.
+     */
+    public PercentileRanksBucketPipelineAggregationBuilder(StreamInput in)
+        throws IOException {
+        super(in, NAME);
+        percents = in.readDoubleArray();
+        keyed = in.readBoolean();
+    }
+
+    @Override
+    protected void innerWriteTo(StreamOutput out) throws IOException {
+        out.writeDoubleArray(percents);
+        out.writeBoolean(keyed);
+    }
+
+    /**
+     * Get the percentages to calculate percentiles ranks for in this aggregation
+     */
+    public double[] getPercents() {
+        return percents;
+    }
+
+    /**
+     * Set the percentages to calculate percentiles ranks for in this aggregation
+     */
+    public PercentileRanksBucketPipelineAggregationBuilder setPercents(double[] percents) {
+        if (percents == null) {
+            throw new IllegalArgumentException("[percents] must not be null: [" + name + "]");
+        }
+        for (Double p : percents) {
```

Comment:
Ditto comment above about defaults, no need for validation here for ranks

Reply:
got it. will accomodate this.